app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -360,64 +227,6 @@ class AudioLibraryAdapter(
         startActivity(context, intent, null)
     }
 
-    private fun setStoppedView(
-        playTimer: Chronometer, remainingTimer: Chronometer,
-        playPauseButton: AppCompatImageButton, isPause: Boolean
-    ) {
-        playTimer.stop()
-        remainingTimer.stop()
-        if (isPause) {
-            isPlayerPaused = true
-        } else {
-            isPlayerStopped = true
-        }
-        playPauseButton.setImageResource(android.R.drawable.ic_media_play)
-    }
-
-    private fun setPlayView(
-        playTimer: Chronometer, remainingTimer: Chronometer,
-        playBar: SeekBar, movePlayBarThread: Runnable,
-        playPauseButton: AppCompatImageButton
-    ) {
-        playTimer.start()
-        remainingTimer.start()
-        isPlayerPaused = false
-        isPlayerStopped = false
-        updatePlayBar(
-            playBar,
-            movePlayBarThread,
-            mediaPlayer!!.duration,
-            mediaPlayer!!.currentPosition
-        )
-        playPauseButton.setImageResource(android.R.drawable.ic_media_pause)
-    }
-
-    private fun handlePlayBarClick(
-        position: Int,
-        playTimer: Chronometer,
-        remainingTimer: Chronometer,
-        playPauseButton: AppCompatImageButton,
-        playBar: SeekBar,
-        movePlayBarThread: Runnable
-    ) {
-        if (isPlayerStopped) {
-            resetRecordPlayer(position, playTimer, remainingTimer, playPauseButton, playBar)
-        }
-
-        if (!mediaPlayer!!.isPlaying) {
-            mediaPlayer?.start()
-
-            if (!isPlayerPaused) {
-                // Reset the play timer
-                playTimer.base = SystemClock.elapsedRealtime()
-            }
-            setPlayView(playTimer, remainingTimer, playBar, movePlayBarThread, playPauseButton)
-        } else {
-            mediaPlayer?.pause()
-            setStoppedView(playTimer, remainingTimer, playPauseButton, true)
-        }
-    }
-
```

Comment:
So if I understand well, we created a blindlyMediaPlayer that extends MediaPlayer and that contains all the fun nedded. Is that correct ?

Reply:
Yep exactly